src/observableApiConfig.ts

```diff
@@ -91,14 +93,15 @@ export async function getDeployConfig(
 
 export async function setDeployConfig(
   sourceRoot: string,
+  requestedDeployConfigPath: string | null = null,
   newConfig: DeployConfig,
   effects: ConfigEffects = defaultEffects
 ): Promise<void> {
-  const dir = op.join(effects.cwd(), sourceRoot, ".observablehq");
-  const deployConfigPath = op.join(dir, "deploy.json");
+  const deployConfigPath =
+    requestedDeployConfigPath ?? op.join(effects.cwd(), sourceRoot, ".observablehq", "deploy.json");
   const oldConfig = (await getDeployConfig(sourceRoot)) || {};
```

Comment:
Also the `|| {}` has no effect? `getDeployConfig` can’t return a falsey value.

Reply:
Yeah, I'm not sure what that was about. Blame says it was added six months ago, maybe that was from the previous version of the deploy configuration? I'll remove it.

Reply:
Ya I imagine at some point `getDeployConfig` returned `null` if there was no existing config, rather than materializing the default config. Harmless but fixed now though!